### PR TITLE
Changed the peeling mechanism

### PR DESCRIPTION
**Pull request by MrSupiri**
#66 happened because I added extra labels to the first match statement and it will always go true, I fixed that with this PR.

I also updated the match peeler to test match statements individually, so now the results would look exactly as we want for the samples given in Issue #66,

image

but I am not sure how it will act with the exclude statements. In banzai docs it says
> Match statements are evaluated in the order they are defined and processed only until the first matching select or exclude rule applies.

if we go with the previous approach (slicing till `N`) and it will look like this
image

this is because it only tests till the first valid one once that valid one is hit it will forward logs even though there are invalid labels.

### CODE REVIEW

**Comment by MrSupiri** [comment]
For filters when slicing is used it's the other way around, It will fail everything after the first failure even though things below are passing

image

image

File: controllers/cleanup.go [hunk 1]
```diff
@@ -107,17 +107,21 @@ func (r *FlowTestReconciler) cleanUpOutputResources(ctx context.Context) error {
...
 		return err
 	}
-	for _, flowTest := range flowTests.Items {
-		if flowTest.Status.Status != loggingplumberv1alpha1.Completed {
-			logger.V(1).Info("unfinished flowtest found, skipping cleanUpOutputResources")
-			return nil
+	if len(flowTests.Items) > 1 {
+		for _, flowTest := range flowTests.Items {
+			if flowTest.Status.Status != loggingplumberv1alpha1.Completed && !flowTest.ObjectMeta.DeletionTimestamp.IsZero() {
+				logger.V(1).Info("unfinished flowtest found, skipping cleanup of log-aggregator")
```
Comment (paynejacob) [inline controllers/cleanup.go:117]: this should be debug level

File: controllers/cleanup.go [hunk 2]
```diff
@@ -161,6 +167,7 @@ func (r *FlowTestReconciler) deleteResources(ctx context.Context, finalizerName
 	// remove our finalizer from the list and update it.
 	controllerutil.RemoveFinalizer(&flowTest, finalizerName)
 	if err := r.Update(ctx, &flowTest); err != nil {
+		logger.Error(err, "failed to remove the finalizer")
```
Comment (paynejacob) [inline controllers/cleanup.go:170]: we should include the flowtest id here

File: controllers/cleanup.go (hunk 1 above)
Comment (MrSupiri) [reply controllers/cleanup.go:117]: V(1) is debug
V(-1) is warning

File: controllers/cleanup.go (hunk 2 above)
Comment (MrSupiri) [reply controllers/cleanup.go:170]: logger append the context automatically

**Review by paynejacob — APPROVED** [review]

File: controllers/provision.go [hunk 3]
```diff
@@ -189,7 +189,10 @@ func (r *FlowTestReconciler) deploySlicedFlows(ctx context.Context, extraLabels
 
 			targetFlow.Spec.GlobalOutputRefs = []string{targetOutput.ObjectMeta.Name}
 
-			targetFlow.Spec.Match = nil
+			// ensure logs are only coming from our simulation pod
+			targetFlow.Spec.Match = []flowv1beta1.ClusterMatch{{
```
[the remark this answers is not in the document]
Comment (MrSupiri) [reply controllers/provision.go:193]: with filter, we can only make sure only it's coming from *one* of the  simulation pods, this match ensure it's only coming from target simulation pod, we can't do this in match tests

**Pull request merged**